src/tab-group.ts

```diff
@@ -54,7 +54,11 @@ export default class TabGroup extends LitElement {
       @keydown=${this.#onComponentKeydown}
       ${ref(this.#componentElementRef)}
     >
-      <div class="tab-container" data-test="tab-container">
+      <div
+        class="tab-container"
+        data-test="tab-container"
+        ${onResize(this.#onTabContainerResize.bind(this))}
```

Comment:
Moved the resize handling to the container instead. Doing so ensures that `#onTabContainerResize()` only gets called on an actual resize of the element. Either by dragging your window, or constraining the width of the TabGroup.

Before this would trigger in many other cases, such as updating the Tab label, as that's _technically_ a resize. But it's pretty confusing that happens unless you know about it. I think this is more explicit and clear, especially with the other changes.